block-explorer/v3.yml

```diff
@@ -0,0 +1,589 @@
+          schema:
+            type: string
+        - name: limit
+          in: query
+          description: Limit (size) of the response items
+          required: false
+          schema:
+            type: integer
+        - name: search_after
+          in: query
+          description: ISOString to search after given timestamp. Please keep in mind that it's mutually exclusive with search_before.
+          required: false
+          schema:
+            type: string
+            title: ISOString
+        - name: search_before
+          in: query
+          description: ISOString to search before given timestamp. Please keep in mind that it's mutually exclusive with search_after.
+          required: false
+          schema:
+            type: string
+            title: ISOString
+      responses:
+        200:
+          description: Transactions from the given global snapshot
+          content:
+            'application/json':
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Transaction'
+        404:
+          description: Not found
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/NotFound'
+  /global-snapshot/{ordinal}/transaction:
+    get:
+      tags:
+        - Transaction
+      summary: Get transactions for the given global snapshot
+      parameters:
+        - name: ordinal
+          in: path
+          description: Ordinal of the global snapshot to return transactions from
+          required: true
+          schema:
+            type: string
+        - name: limit
+          in: query
+          description: Limit (size) of the response items
+          required: false
+          schema:
+            type: integer
+        - name: search_after
+          in: query
+          description: ISOString to search after given timestamp. Please keep in mind that it's mutually exclusive with search_before.
+          required: false
+          schema:
+            type: string
+            title: ISOString
+        - name: search_before
+          in: query
+          description: ISOString to search before given timestamp. Please keep in mind that it's mutually exclusive with search_after.
+          required: false
+          schema:
+            type: string
+            title: ISOString
+
+      responses:
+        200:
+          description: Transactions from the given global snapshot
+          content:
+            'application/json':
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Transaction'
+        404:
+          description: Not found
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/NotFound'
+  /address/{address}/transaction:
+    get:
+      tags:
+        - Transaction
+      summary: Get transactions for the given address
```

Comment:
Should we mention in the api about the order of transactions? Also maybe let the client decide the order (ascending, descending)? I guess we will have some default order. And with the limit I guess we should also allow to set the order so that the client can take last 5 or first 5 transaction e.g.

Reply:
I think we don't need to mention that explicitly in the API. `search_after` and `search_before` should be used. we may try to figure out whether `sort` and `limit` can be used separately but I'd keep that as an optimization if needed